core/src/main/java/org/apache/accumulo/core/metadata/StoredTabletFile.java

```diff
@@ -60,6 +65,22 @@ public Text getMetaUpdateDeleteText() {
     return new Text(getMetaUpdateDelete());
   }
 
+  public TabletFile getTabletFile() {
+    return tabletFile;
+  }
+
+  public TableId getTableId() {
+    return tabletFile.getTableId();
+  }
+
+  public String getPathStr() {
+    return tabletFile.getPathStr();
+  }
+
+  public Text getMetaInsertText() {
+    return tabletFile.getMetaInsertText();
+  }
```

Comment:
I think this method should be deleted because it normalizes the path. Its only used in one place and where its used we are updating existing files in the metadata table, so do not want to normalize.

```suggestion
```

Reply:
So for this I have the same comment as I had with adding a putFile() method that takes a StoredTabletFile.

In this existing code this is used and we always normalize where this is [called](https://github.com/apache/accumulo/blob/0017fd8949cc1bc7253b8e90313d4fd1b71e2241/server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java#L261).

So, we are normalizing today even for StoredTabletFile and I'm wondering what happens if we don't. Is this actually bug and we are using the wrong value but it's not showing up as the normalized path usually equals the existing path?

Reply:
That seems like an existing bug to me.  Making these types independent makes it harder to have these bugs. In the older code even with the separate type for StoredTabletFile it seems it was still really hard to use correctly (like one had to know no to call getMetaInsertText() when doing a metadata update).

> So, we are normalizing today even for StoredTabletFile and I'm wondering what happens if we don't.

It seems like we are normalizing.  I suspect normally it does not matter because the normalization ends up being the same as what is stored.  If the normalization is ever not the same it will cause very subtle problems.

In the elasticity branch this will eventually be less of a problem with conditional update because a condition can be added ot the mutation to require the exact files tbeing updated to currently exists in the metadata table.  With the existing unconditional mutations, need to ensure the string matches exactly what is stored in the metadata table otherwise data could unintentionally be duplicate or not deleted.

Reply:
Once #3432 is merged I will update this PR to remove this.

Reply:
This has now been removed.